sleap/gui/app.py

```diff
@@ -325,8 +325,18 @@ def _create_video_player(self):
         self.player.seekbar.selectionChanged.connect(lambda: self.updateStatusMessage())
         self.setCentralWidget(self.player)
 
-        def switch_frame(video):
-            """Jump to last labeled frame"""
+        def switch_frame(self, video):
+            """Maintain the same frame or jump to the last labeled frame."""
+            # get current frame index
+            current_frame_idx = self.state["frame_idx"]
+            # check to see if new video has enough frames
+            if video.num_frames >= current_frame_idx:
```

Comment:
After manual testing, this actually needs to be `>` not `>=` due to the 1-indexing of `state["frame_idx"]`.